apollo-router/src/plugins/rhai/mod.rs

```diff
@@ -82,6 +83,9 @@ type SharedMut<T> = rhai::Shared<Mutex<Option<T>>>;
 
 pub(crate) const RHAI_SPAN_NAME: &str = "rhai_plugin";
 
+static RHAI_GLOBAL_VARIABLES: Lazy<Arc<Mutex<Map>>> =
+    Lazy::new(|| Arc::new(Mutex::new(Map::new())));
```

Comment:
Maybe it might be interesting to put a `RwLock` instead of a mutex since it's constant

Reply:
fixed: c4b621ae96c64cbf037575cbadc75283b1f4ec98

(I removed the static completely)